src/encoding.rs

```diff
@@ -167,7 +169,12 @@ where
         let byte = buf.get_u8();
         value |= u64::from(byte & 0x7F) << (count * 7);
         if byte <= 0x7F {
-            return Ok(value);
+            // check for u64::MAX overflow
+            if count == 9 && byte >= 2 {
```

Comment:
Decided to add this check here in case moving it outside the first `if` caused some performance impact.